juju/juju-restore
# Unpack the backup file and extract metadata

## Description of change

This implements opening the backup file provided to restore and extracting the metadata from it. Checking that against the connected controller will be in a followup PR.

## QA steps

Run juju-restore with an invalid backup file path, or a path to a file that isn't a valid backup - it'll fail.
With a valid backup file, if you specify a non-default value for the --temp-root option you'll be able to see the backup file unpacked at that location while `juju-restore` is waiting for input, but it will be cleaned up when the program finishes.
